src/java.base/share/classes/java/lang/String.java

```diff
@@ -4535,9 +4541,12 @@ void getBytes(byte[] dst, int srcPos, int dstBegin, byte coder, int length) {
     String(AbstractStringBuilder asb, Void sig) {
         byte[] val = asb.getValue();
         int length = asb.length();
+        // To avoid surprises due to data races (which would either truncate or throw an exception)
+        // we should check that length <= val.length up front
+        checkOffset(length, val.length);
```

Comment:
I agree a check is needed here but I assume using checkOffset means that SB::toString could fail with SIOOBE. I wonder if Math.min(length, val.length) would be better here.

Reply:
Ok. That keeps behavior consistent for most cases and removes a path where we can fail with SIOOBE in the existing code (down `StringUTF16::compress`).